lib/mix/lib/mix/tasks/deps.update.ex

```diff
@@ -61,6 +61,20 @@ defmodule Mix.Tasks.Deps.Update do
         Mix.Dep.Fetcher.all(Mix.Dep.Lock.read(), %{}, fetch_opts)
 
       rest != [] ->
+        invalid =
+          Enum.find(rest, fn name ->
+            !String.match?(name, ~r/^[a-zA-Z0-9_]+$/)
```

Comment:
Do we validate dependency names somewhere? If so, where is it listed? Can we reuse the logic?

Reply:
I will take a look!

Reply:
This may actually have been a mistake on my part? I'm actually able to put deps like `{:"1foo", ...}` into `mix.exs`. Using `mix new 1foo` shows an error about the name, but it seems like hex is actually about to let me publish a package with that name when I modify it manually:

![CleanShot 2024-10-30 at 09 22 10@2x](https://github.com/user-attachments/assets/53d4878e-52c8-4ef2-a824-43a2fc2f94c7)

So options going from here are:
1. close this PR.
2. validate app names more aggressively in other places (pretty much an unrelated effort to this PR)
3. simplify this check to instead do something along the lines of allowing dep names to match if they have a prefixed `:` symbol. This would prevent the confusing scenario, but might not be worth the effort.

Thoughts?